features/lorawan/LoRaWANStack.cpp

```diff
@@ -1067,8 +1063,8 @@ lorawan_status_t LoRaWANStack::state_controller(device_states_t new_state)
             process_shutdown_state(status);
             break;
         default:
-            tr_debug("state_controller: Unknown state!");
-            status = LORAWAN_STATUS_SERVICE_UNKNOWN;
+            //Because this is internal function only coding error causes this
+            MBED_ASSERT(false);
```

Comment:
But in release profile we shouldn't return LORAWAN_STATUS_OK either. So I would keep the original code but also have the assert.

Reply:
Please keep the original code here. Or just add assert at the end after the original code. As @kivaisan mentioned, release build would strip off asserts.